Fix How-to Commit docs

bereng opened the pull request.

bereng committed:
Fix How-to Commit docs

michaelsembwever commented on doc/source/development/how_to_commit.rst:
```diff
@@ -84,6 +84,21 @@ On trunk:
+  
+  ::
+
+      <One sentence description, usually Jira title and CHANGES.txt summary>
```
```suggestion
      <One sentence description, usually Jira title or CHANGES.txt summary>
```

bereng replied:
added

michaelsembwever commented on doc/source/development/how_to_commit.rst:
```diff
@@ -23,66 +23,81 @@ If you are a committer, feel free to pick any process that works for you - so lo
+      patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####
+      
+      
+          Co-authored-by: Name1 <email1>
+          Co-authored-by: Name2 <email2>
```
```suggestion
Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>
```

(no idea if it will work with the leading space)

bereng replied:
done

ekaterinadimitrova2 reviewed (approved):
Only wondering whether we can put a sentence around the co-author thing to make a note that it is possible to add an author or even to change one with the following command:
git commit --author="John Doe <john@doe.org>"

bereng commented:
So I added your `git author` sentence @ekaterinadimitrova2 as it sounds like a good idea. But then I removed it bc it sounded confusing in my head. You can _update_ the author info but you can't _add_ multiple authors i.e. So this is more like a git command than how to format a commit message to GH stds so it parses multiple authors.

They are similar things but not the same hence I removed it. Are you ok with it?

EDIT: just saw your comment you're ok if this sentence is not added :-)

ekaterinadimitrova2 commented:
That is what I meant, they are not the same. The idea was to have just the two things mentioned. 
You can change the author in case the patch is from someone else but you can't use the original sha. 
And also, you can have cases with more than one author
